### PR TITLE
BauraBAL strategy migration script

### PR DESCRIPTION
**Pull request by sajanrajdev**
This PR tackles issue #633 

To post:
```
brownie run scripts/issue/633/migrate_baurabal_strategy.py
```
To run simulation:
```
brownie run scripts/issue/633/migrate_baurabal_strategy.py main true
```

NOTE: As part of the migration process, the naked bb-a-usd accumulated for the current strat will also be emitted.

### CODE REVIEW

File: scripts/issue/633/migrate_baurabal_strategy.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        == auraBal_strat_current.claimRewardsOnWithdrawAll()
+    )
+    assert (
+        auraBal_strat_new.balEthBptToAuraBalMinOutBps()
+        == auraBal_strat_current.balEthBptToAuraBalMinOutBps()
+    )
+
+    ## Harvest current strat
+    prev_bbausd_strat_balance = bb_a_usd.balanceOf(AURABAL_STRAT_CURRENT)
+    auraBal_strat_current.harvest()
```
Comment (gosuto-inzasheru) [inline scripts/issue/633/migrate_baurabal_strategy.py:46]: im getting a revert here?

File: scripts/issue/633/migrate_baurabal_strategy.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    assert auraBal_strat_current.balanceOf() == 0
+
+    # setStrategy
+    assert auraBal_strat_current.want() == auraBal_strat_new.want() # Confirm strategy
+    auraBal_vault.setStrategy(AURABAL_STRAT_NEW)
+    assert auraBal_vault.strategy() == auraBal_strat_new.address
+
+    # Earn to new strat
+    prev_available = auraBal_vault.available()
+    prev_strat_balance = auraBal_strat_new.balanceOf()
```
Comment (shuklaayush) [inline scripts/issue/633/migrate_baurabal_strategy.py:72]: I guess this should be 0 anyways

File: scripts/issue/633/migrate_baurabal_strategy.py (hunk 1 above)
Comment (shuklaayush) [reply scripts/issue/633/migrate_baurabal_strategy.py:46]: Working for me. Can you share the trace?

**Review by shuklaayush — APPROVED** [review]

**Review by GalloDaSballo — APPROVED** [review]
Code LG, Script will migrate funds to new strategy

File: scripts/issue/633/migrate_baurabal_strategy.py (hunk 2 above)
Comment (sajanrajdev) [reply scripts/issue/633/migrate_baurabal_strategy.py:72]: Yes, it should be, decided to do it like this for code reusability

**Pull request merged**